reframe/core/meta.py

```diff
@@ -417,31 +437,22 @@ class attribute. This behavior does not apply when the assigned value
         is not allowed. This would break the parameter space internals.
         '''
 
-        # Set the value of a variable (except when the value is a descriptor).
+        # Try to treat `name` as variable
         try:
-            var_space = super().__getattribute__('_rfm_var_space')
-            if name in var_space:
-                if not hasattr(value, '__get__'):
-                    var_space[name].define(value)
-                    return
-                elif not var_space[name].field is value:
-                    desc = '.'.join([cls.__qualname__, name])
-                    raise ReframeSyntaxError(
-                        f'cannot override variable descriptor {desc!r}'
-                    )
-
-        except AttributeError:
-            pass
+            cls.setvar(name, value)
+            return
+        except AttributeError as e:
+            '''`name` is not a variable'''
 
         # Catch attempts to override a test parameter
         try:
             param_space = super().__getattribute__('_rfm_param_space')
             if name in param_space.params:
                 raise ReframeSyntaxError(f'cannot override parameter {name!r}')
-
         except AttributeError:
-            pass
+            '''Test is not parametrized'''
```

Comment:
Just for consistency with the `__getattr__` above.
```suggestion
            '''Catch early access attempt to the parameter space.'''
```